torch/distributed/distributed_c10d.py

```diff
@@ -929,11 +947,39 @@ def all_reduce(tensor,
         Async work handle, if async_op is set to True.
         None, if not async_op or if not part of the group
 
+    Examples:
+        >>> # All tensors below are of torch.int64 type.
+        >>> # We have 2 process groups, 2 ranks.
+        >>> tensor = torch.arange(2, dtype=torch.int64) + 1 + 2 * rank
+        >>> tensor
+        tensor([1, 2]) # Rank 0
+        tensor([3, 4]) # Rank 1
+        >>> dist.all_reduce(tensor, op=ReduceOp.SUM)
+        >>> tensor
+        tensor([4, 6]) # Rank 0
+        tensor([4, 6]) # Rank 1
+
+        >>> # All tensors below are of torch.cdouble type.
+        >>> # We have 2 process groups, 2 ranks.
+        >>> tensor = torch.tensor([1+1j, 2+2j], dtype=torch.cdouble) + 2 * rank * (1+1j)
```

Comment:
actually maybe it's best to let it be `torch.cfloat` because that's the default complex type if the default dtype is set to torch.float. This is consistent with the above example since default int dtype is `torch.long` or `torch.int64`.
If we want to use `torch.cdouble`, all the following prints of tensor, would look like:
`tensor([1.+1.j, 2.+2.j], dtype=torch.complex128)`

Reply:
Makes sense. done